src/axom/slam/RelationSet.hpp

```diff
@@ -154,14 +171,11 @@ class RelationSet final
    * \return  An OrderedSet containing the elements in the row.
    * \pre  0 <= pos1 <= set1.size()
    */
-  const OrderedSetType getElements(PositionType s1) const override
-  {
-    return (*m_relation)[s1];
-  }
+  SubsetType getElements(PositionType s1) const { return (*m_relation)[s1]; }
 
-  ElementType at(PositionType pos) const override
+  ElementType at(PositionType pos) const
   {
-    verifyPositionImpl(pos);
+    verifyPosition(pos);

```

Comment:
@publixsubfan -- please verify that there's no performance cost associated with this change.

I intentionally added `verifyPositionImpl()` as a non-virtual function to allow the compiler to inline the call (especially in release builds, where this gets optimized away). `verifyPosition` was a virtual function and the compiler was not able to optimize it.

Perhaps this concern goes away with the new implementation. In which case, I'm strongly in favor of the cleaner code!

Reply:
I think this is non-virtual when used with a concrete policy. To address the virtual case, I think we can directly invoke the function nonvirtually, i.e. `this->RelationSet::verifyPosition(pos)`